Color expiration indicator correctly in dark theme

RiseT opened the pull request:
<!-- You can remove this first section if you have contributed before -->
### First time contributor checklist
<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist
<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description
<!--
Describe briefly what your pull request proposes to fix. Especially if you have more than one commit, it is helpful to give a summary of what your contribution as a whole is trying to solve.
Also, please describe shortly how you tested that your fix actually works.
-->

Fixes #5759

Changes the color of the expiration indicator (disappearing messages) from black to the color used for the other message indicators in the dark theme.

// FREEBIE

moxie0 commented:
can you post before and after screenshots for both light and dark themes? thanks!

FeuRenard commented:
### Screenshots
before
![device-2016-10-09-185935](https://cloud.githubusercontent.com/assets/16167751/19222187/d390eae2-8e52-11e6-96bd-9d4a752d5176.png)
after
![device-2016-10-09-185737](https://cloud.githubusercontent.com/assets/16167751/19222186/d3687dc8-8e52-11e6-82e1-8757d462925a.png)
before
![device-2016-10-09-190306](https://cloud.githubusercontent.com/assets/16167751/19222220/8da8041a-8e53-11e6-8d93-671cef9a66cd.png)
after
![device-2016-10-09-190506](https://cloud.githubusercontent.com/assets/16167751/19222221/8fbe99f8-8e53-11e6-8a98-7aca1af8b4b7.png)

moxie0 commented:
The light theme doesn't look quite right in the after configuration (hourglass is lighter than the lock and the checks).  Sadly the dark theme looks off both in the before and after!

RiseT commented:
I see. Closing for now.

haffenloher commented:
@RiseT indeed. On that screen shot the colors are b7b7b7 for the lock and 898989 for the check mark

RiseT commented:
Ugh. This is getting complicated. I've played with this a bit, and I think the coloring discrepancies of the lock are due to the Android version used.

Example:
Take a look at the "before" screenshot  for the black theme posted by @FeuRenard. This is from Android 4.4.4:
![d390eae2-8e52-11e6-96bd-9d4a752d5176](https://cloud.githubusercontent.com/assets/12624674/19225315/6d466b52-8e9a-11e6-94a2-c0edf05a5ff7.png)

Now take a look at a "before" screenshot I've taken on a Nexus 5X with Android 7.0:
![48800c26-8d5d-11e6-84a4-510331d9f283](https://cloud.githubusercontent.com/assets/12624674/19225317/7b39d910-8e9a-11e6-8174-095713424b48.png)

Notice that the color of the lock is much lighter on Android 4.4.4.
I've confirmed this on an actual Android 4.4.4 device (one I've plugged into Android Studio) myself.

I wonder how this PR's changes look like on Android 6 or 7 (on a screenshot without too much image compression). Unfortunately, I cannot check this out myself as the emulator of my Android Studio keeps crashing currently.

RiseT commented:
The secure indicator/lock icon somehow doesn't respect the alpha channel value, i. e. opacity at all on Android 4.4.4. That's why it's lighter - opacity is always 100%.

moxie0 commented:
the original image probably isn't colored correctly

RiseT commented:
@moxie0 Thanks for the pointer.

Btw - I've noticed that [here](https://github.com/WhisperSystems/Signal-Android/blob/549e45e2d6d511571bff7b3fac9c6c6a7b0bcdbf/res/values/themes.xml#L202) (light theme) and [here](https://github.com/WhisperSystems/Signal-Android/blob/549e45e2d6d511571bff7b3fac9c6c6a7b0bcdbf/res/values/themes.xml#L323) (dark theme) Android studio says that it cannot resolve the symbol. Is there something missing?

haffenloher commented:
@RiseT 

> I've noticed that here (light theme) and here (dark theme) Android Studio says that it cannot resolve the symbol.

Can't confirm this :/ If I press Ctrl+B (Go To Declaration) on *@style/BetterPickersDialogFragment(.Light)*, Android Studio correctly jumps to the declaration in the com.doomonafireball.betterpickers package

RiseT commented:
@haffenloher  Mh. When I do the same, it says 'Cannot find the declaration to go'. Never mind. If it's working at your place, then it's fine. Still trying to find my way through Android Studio. Although I'm not aware of changing something there, I'm a noob.

haffenloher commented:
@RiseT updated Android Studio from 2.1 to 2.2 - getting the same error message now :confused: let's move over to [the forums](https://whispersystems.discoursehosting.net/t/emulator-in-android-studio-crashing-when-touching-a-picker-element/547/4)...

RiseT commented:
I've found the reason why the hourglass looks somewhat weird.

As ic_hourglass_empty_white_18dp.png and ic_hourglass_full_white_18dp.png are displayed in front of each other as 2 layers, when setting alpha/opacity to e. g. 75%, the opacity of both images adds up whereever there is an overlap between the 2 images.

Currently, the first image displays an empty hourglass and the second image a full hourglass - i. e. the "glass" part of the hourglass is overlapping.

In order to make this look as intended, the second image should **only** contain the "sand" of the hourglass, not the whole full hourglass. This eliminates the overlap and thus any pixels with weird alpha values.

2-4601 commented:
can't see them pics, tried chrome and firefox

RiseT commented:
@2-4601 I'm using Firefox right know. When I click onto one of the pics, it opens in another tab with a dark background.

rainerzufall commented:
works with firefox when clicked. not on opera or chrome

RiseT commented:
That's what I have so far.

This is more work than I initially thought (especially given I know absolutely nothing about Android development or graphics editing). As all elements have to optically match, one thing leads to another.

**Changes:**
- Modified coloring code & all image resources of hourglass
- Modified all image resources for lock
- Added semi-transparency for some stuff like Conversation Item Date

**Caveats:**
- The screenshots were taken with Android 7. There's still that problem with the weird coloring/opacity of the lock on Android 4.4.4. My uneducated guess regarding the cause for this is that it has API level 19, while the corresponding coloring code for the lock is using ```Android:tintMode```, which has been introduced with API level 21 (at least that's what Android Studio says...).
- The vertical alignment of all of date, hourglass, lock, etc seems a bit uneven to me, especially on the "sent" side. Imho, all of these should be aligned to the bottom, i. e. the horizontal position of the lowest pixel should be identical (don't know how to do this, though).

# Light theme
## Before
![light master screenshot_20161021-160407 ano](https://cloud.githubusercontent.com/assets/12624674/19603302/f28934c4-97b0-11e6-9f10-e28552388544.png)

## After
![light screenshot_20161021-150512 ano](https://cloud.githubusercontent.com/assets/12624674/19603325/ff5ada5e-97b0-11e6-9341-f6b00508efe8.png)

# Dark theme
## Before
![dark master screenshot_20161021-160407 ano](https://cloud.githubusercontent.com/assets/12624674/19603333/0bfb480c-97b1-11e6-8ef5-5f0d9d8a6e68.png)

## After
![dark screenshot_20161021-150452 ano](https://cloud.githubusercontent.com/assets/12624674/19603345/18996e36-97b1-11e6-9278-b4063162c12e.png)
